core/src/main/java/org/elasticsearch/ingest/PipelineStore.java

```diff
@@ -156,8 +158,12 @@ public void put(ClusterService clusterService, PutPipelineRequest request, Actio
             // validates the pipeline and processor configuration before submitting a cluster update task:
             Map<String, Object> pipelineConfig = XContentHelper.convertToMap(request.getSource(), false).v2();
             constructPipeline(request.getId(), pipelineConfig);
+        } catch (ConfigurationPropertyException e) {
+            listener.onResponse(new WritePipelineResponse(new PipelineFactoryError(e)));
+            return;
         } catch (Exception e) {
-            throw new IllegalArgumentException("Invalid pipeline configuration", e);
+            listener.onResponse(new WritePipelineResponse(new PipelineFactoryError("Invalid pipeline configuration")));
```

Comment:
we should include `e` here, otherwise we lose the cause of the configuration error.